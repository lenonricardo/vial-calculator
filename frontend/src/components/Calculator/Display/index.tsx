import {
  Wrapper,
  Header,
  CustomHistoryIcon,
  Result,
  CustomBackspaceIcon
} from './styles'

interface DisplayProps {
  result: string
  onCancelEntry: () => void
  onShowHistory: () => void
}

export default function Display({
  result,
  onCancelEntry,
  onShowHistory
}: DisplayProps) {
  return (
    <Wrapper>
      <Header>
        <CustomHistoryIcon onClick={() => onShowHistory()}></CustomHistoryIcon>
      </Header>

      <Result>{result || 0}</Result>
      <CustomBackspaceIcon onClick={() => onCancelEntry()} />
    </Wrapper>
  )
}
