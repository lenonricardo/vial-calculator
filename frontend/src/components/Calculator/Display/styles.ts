import styled from 'styled-components'
import BackspaceIcon from '@mui/icons-material/Backspace'
import HistoryIcon from '@mui/icons-material/History'

export const Wrapper = styled.div`
  width: var(--primary-width);
  border-radius: 1.5rem 1.5rem 0 0;
  display: flex;
  align-items: flex-end;
  justify-content: flex-end;
  flex-direction: column;
`

export const CustomBackspaceIcon = styled(BackspaceIcon)`
  && {
    font-size: var(--default-font-size);
    margin-right: 10px;
    color: var(--theme-secondary);
    cursor: pointer;
  }
`

export const Result = styled.div`
  color: var(--font-primary-color);
  font-size: var(--result-font-size);
  margin-right: 10px;
  white-space: nowrap;
  overflow-x: auto;
  width: 430px;
  text-align: right;
`

export const Header = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 100%;
  padding: 10px;
`

export const CustomHistoryIcon = styled(HistoryIcon)`
  && {
    font-size: var(--default-font-size);
    margin-right: 10px;
    color: var(--theme-secondary);
    cursor: pointer;
  }
`
