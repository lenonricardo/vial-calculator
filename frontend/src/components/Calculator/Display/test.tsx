import { render } from '@testing-library/react'

import Display from '.'

describe('<Display />', () => {
  it('should render the component', () => {
    const onCancelEntry = jest.fn()
    const onShowHistory = jest.fn()

    const { container } = render(
      <Display
        result="123"
        onCancelEntry={onCancelEntry}
        onShowHistory={onShowHistory}
      />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
