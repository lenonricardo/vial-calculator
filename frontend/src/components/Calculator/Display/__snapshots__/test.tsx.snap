// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> should render the component 1`] = `
.c0 {
  width: var(--primary-width);
  border-radius: 1.5rem 1.5rem 0 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4.c4 {
  font-size: var(--default-font-size);
  margin-right: 10px;
  color: var(--theme-secondary);
  cursor: pointer;
}

.c3 {
  color: var(--font-primary-color);
  font-size: var(--result-font-size);
  margin-right: 10px;
  white-space: nowrap;
  overflow-x: auto;
  width: 430px;
  text-align: right;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  padding: 10px;
}

.c2.c2 {
  font-size: var(--default-font-size);
  margin-right: 10px;
  color: var(--theme-secondary);
  cursor: pointer;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c2 css-i4bv87-MuiSvgIcon-root"
      data-testid="HistoryIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z"
      />
    </svg>
  </div>
  <div
    class="c3"
  >
    123
  </div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c4 css-i4bv87-MuiSvgIcon-root"
    data-testid="BackspaceIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M22 3H7c-.69 0-1.23.35-1.59.88L0 12l5.41 8.11c.36.53.9.89 1.59.89h15c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-3 12.59L17.59 17 14 13.41 10.41 17 9 15.59 12.59 12 9 8.41 10.41 7 14 10.59 17.59 7 19 8.41 15.41 12 19 15.59z"
    />
  </svg>
</div>
`;
